src/main/java/org/sagebionetworks/bridge/services/StudyActivityEventService.java

```diff
@@ -0,0 +1,213 @@
+    private static final String CREATED_ON_FIELD = CREATED_ON.name().toLowerCase();
+
+    private StudyActivityEventDao dao;
+    private AppService appService;
+    private AccountService accountService;
+    
+    @Autowired
+    final void setStudyActivityEventDao(StudyActivityEventDao dao) {
+        this.dao = dao;
+    }
+    @Autowired
+    final void setAppService(AppService appService) {
+        this.appService = appService;
+    }
+    @Autowired
+    final void setAccountService(AccountService accountService) {
+        this.accountService = accountService;
+    }
+    
+    DateTime getCreatedOn() { 
+        return DateTime.now();
+    }
+
+    /**
+     * Only custom events can be deleted (if they are mutable). Other requests 
+     * are silently ignored. 
+     */
+    public void deleteCustomEvent(StudyActivityEventRequest request) {
+        checkNotNull(request);
+        checkArgument(request.getObjectType() == CUSTOM);
+
+        App app = appService.getApp(request.getAppId());
+        request.customEvents(app.getCustomEvents());
+        
+        StudyActivityEvent event = request.toStudyActivityEvent();
+        
+        Validate.entityThrowingException(DELETE_INSTANCE, event);
+        
+        StudyActivityEvent mostRecent = dao.getRecentStudyActivityEvent(
+                event.getUserId(), event.getStudyId(), event.getEventId());
+
+        if (request.getUpdateType().canDelete(mostRecent, event)) {
+            dao.deleteCustomEvent(event);
+        }
+    }
+    
+    public void publishEvent(StudyActivityEventRequest request) {
+        checkNotNull(request);
+
+        App app = appService.getApp(request.getAppId());
+        request.customEvents(app.getCustomEvents());
+        request.createdOn(getCreatedOn());
+        
+        StudyActivityEvent event = request.toStudyActivityEvent();
+        
+        Validate.entityThrowingException(CREATE_INSTANCE, event);
+        
+        StudyActivityEvent mostRecent = dao.getRecentStudyActivityEvent(
+                request.getUserId(), request.getStudyId(), event.getEventId());
+        
+        if (request.getUpdateType().canUpdate(mostRecent, event)) {
+            dao.publishEvent(event);
+            createAutomaticCustomEvents(app, request);
+        }
+    }
+    
+    public ResourceList<StudyActivityEvent> getRecentStudyActivityEvents(String appId, String userId, String studyId) {
+        checkNotNull(userId);
+        checkNotNull(studyId);
+
+        Account account = accountService.getAccountNoFilter(AccountId.forId(appId, userId))
+                .orElseThrow(() -> new EntityNotFoundException(Account.class));
+
+        List<StudyActivityEvent> events = dao.getRecentStudyActivityEvents(userId, studyId);
+        events.add(new StudyActivityEvent(CREATED_ON_FIELD, account.getCreatedOn()));
+        events.add(makeStudyStartDate(events, account.getCreatedOn()));
+        
+        return new ResourceList<>(events); 
```

Comment:
Do we need the created_on timestamp to be written back to the database?
Also talk to Dwayne about study_start_date.

Reply:
Answer: no, the APIs point at the global system and I haven't removed anything there about persisting a created_on timestamp from the Account.